ipfsApi/multipart.py

```diff
@@ -386,37 +386,43 @@ def _prepare(self):
                 # remove leading / or \ if it is present
                 if short_name.startswith(os.sep):
                     short_name = short_name[1:]
-                # add the file to those being sent
-                names.append(('files', (short_name,
-                                        open(filepath, 'rb'),
-                                        'application/octet-stream')))
+                try:
+                    # add the file to those being sent
+                    names.append(('files', (short_name,
+                                            open(filepath, 'rb'),
+                                            'application/octet-stream')))
+                except OSError:
+                    # File might have disappeared between `os.walk()`
```

Comment:
No, I just looked audited any code that can potentially cause an exception and the rest was experience and probably the fact that I read some stuff on Linux' `inotify` API some time ago.